aten/src/ATen/native/cpu/SoftMaxKernel.cpp

```diff
@@ -218,10 +332,10 @@ inline void _vec_softmax(
   int64_t dim_stride = inner_size;
   int64_t outer_stride = dim_size * dim_stride;
   int64_t grain_size = std::min(internal::GRAIN_SIZE / dim_size, (int64_t)1);
-  int vectorized_step = Vec().size(); // Currently, we only support scalar_t with double or float32
+  int vectorized_step = Vec().size(); // Currently, we only support scalar_t with bfloat16, float32 or double
   TORCH_CHECK(
-    (vectorized_step == 8) || (vectorized_step == 4),
-    "vectorized_step must be 8 with dtype float or 4 with dtype double");
+    (vectorized_step == 16) || (vectorized_step == 8) || (vectorized_step == 4),
```

Comment:
nit: Could it make sense to statically enable these checks dependent on scalar_t?

Reply:
Thanks for the comment. It has been updated with `assert_vectorized_step<dtype>` template.